@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -754,6 +753,10 @@ export class MapControls extends THREE.EventDispatcher {
             this.m_maxTiltAngle
         );
 
+        if (resetZoomState) {
+            this.m_targetedZoom = undefined;
+            this.m_currentZoom = undefined;
```

Comment:
Why do we even have m_currentZoom? Are there situations when it will be different then mapview.zoomLevel?

Reply:
I'm not a MapControls expert, but I think it is used for animating / inertia.

Reply:
Yeah, I would understand m_targetedZoom for the animation

Reply:
Since this is unrelated to this change but a bigger cleanup for MapControls let's keep it for now.

Reply:
m_currentZoom is calculated as:

```
this.currentZoom =
            !this.inertiaEnabled || Math.abs(this.zoomLevelTargeted - this.m_startZoom) < EPSILON
                ? this.zoomLevelTargeted
                : this.easeOutCubic(
                      this.m_startZoom,
                      this.zoomLevelTargeted,
                      Math.min(1, this.m_zoomAnimationTime / this.zoomInertiaDampingDuration)
                  );

```

Do you want to store this directly in the mapView?

Reply:
Agreed it should be cleaned up!

Reply:
I can only say that i also wanted to understand internal state handling on MapControls and real meaning of `currentZoom` but i failed. That's why this patch is minimal so and doesn't try to change big picture of things.